Fixed #24708 -- Handled non-string values in GenericIPAddressField.to_python()

Pradeek opened the pull request:
Handled a case of GenericIPAddressField raising an exception if non
strings are passed in.

timgraham commented on django/db/models/fields/__init__.py:
```diff
@@ -1963,6 +1963,8 @@ def get_internal_type(self):
 
     def to_python(self, value):
+        if value and not isinstance(value, six.string_types):
+            value = smart_text(value)
```
I'd think we may need to use `force_text()` -- won't the next line fail if a lazy instance isn't resolved?

Pradeek replied:
True. Should've checked for that. Will update it.

timgraham commented:
buildbot, add to whitelist.

timgraham commented:
buildbot, add to whitelist.

Pradeek commented:
@timgraham Can you checkout the CI errors? Looks like its unable to checkout the rev?

timgraham commented:
Could you add a test for the lazy case, squash commits, and follow the commit message guidelines as noted in our [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist)?

kezabelle commented on django/db/models/fields/__init__.py:
```diff
@@ -1963,6 +1963,8 @@ def get_internal_type(self):
         return "GenericIPAddressField"
 
     def to_python(self, value):
+        if value and not isinstance(value, six.string_types):
```
You've lost the `value = value.strip()` (line 1223 as of `master` at 8714403614c4dfa37e806db4a6708b8a91a827a4)

Pradeek replied:
Good catch. Fixed it.

timgraham replied:
Could we add something like:
```
if value is None:
    return None
```
to avoid checking `if value:` 3 times?

Pradeek commented:
@timgraham Can you check the CI errors again? I've added the relevant tests, squashed the commits and updated the PR. The tests were passing just before squashing. Now it says `FATAL: Could not checkout 2e0563dfd9130ca77fc41babbad66e54026374ca` Any suggestions?

timgraham commented:
Sometimes those checkout errors happen from time to time, no need to worry about them.

timgraham commented:
'Fixes' -> 'Fixed' in the commit message; otherwise, this looks good.

Pradeek commented:
@timgraham Done.

timgraham commented:
merged in 6123e6134fba627a49ad379edfc20ae0f73bc9ac, thanks!
